src/App.js

```diff
@@ -38,9 +38,14 @@ class App extends React.Component {
     this.getWeatherData();
   };
 
-  getWeatherData = async () => {
-    let response = await axios.get(`${apiUrl}/weatherData`);
-    console.log('response from weather data', response);
+  getWeatherData = async (location) => {
+    const response = await axios.get(`${apiUrl}/weatherData`, {
+      params: {
+        lat: this.state.location.lat,
+        lon: this.state.location.lon,
```

Comment:
You want to use your `location` parameter, and pass `location` in to `getWeatherData(location)` above, because `this.state.location` might not be available.

Try adding a `console.log(this.state.location)` after your `this.setState(...)` to see it's not available yet.